spark/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -41,35 +41,30 @@ import org.apache.spark.sql.types.StructType
  * @since 0.3.0
  */
 class DeltaTable private[tables](
-    @transient private val _df: Dataset[Row],
-    @transient private val table: DeltaTableV2)
+    @transient private val _table: DeltaTableV2,
+    private val alias: Option[String] = None)
   extends DeltaTableOperations with Serializable {
 
-  protected def deltaLog: DeltaLog = {
-    /** Assert the codes run in the driver. */
-    if (table == null) {
-      throw DeltaErrors.deltaTableFoundInExecutor()
-    }
+  // Ensure table's initial snapshot has been captured, for consistent behavior.
```

Comment:
What does capture means? Consistent to what? Maybe a bit more details here would be helpful

Reply:
Something like this?
> WARNING: `DeltaTableV2.initialSnapshot` is lazy, and immutable once triggered. That means it will be stale after DML operations like `DeltaTable.update` or `DeltaTable.delete`. However, those commands don't necessarily trigger the snapshot, which could give subsequent snapshot-using operations the illusion of freshness if their first snapshot access is after the first DML. Illusion, because it wouldn't work a second time. Thus, we trigger the snapshot immediately to at least give consistently stale behavior.